aces/Loris
# Fixes and improvements for the CouchDB_MRI_Importer.

## Brief summary of changes

This PR fixes a bug in the way that the data dictionary entries associated to MRI comments were named. 

A little background first:

Every kind of information that be entered via the MRI feedback panel has to go into the data dictionary. The MRI feedback panel allows a user to enter various types of comments for a given scan. For each of these comment types (listed in table feedback_mri_comment_types), we have:
 
1. A comment status field (column `CommentStatusField` in table `feedback_mri_comment_types`) which can have a fixed set of values. For example, for comments of types 'Dominant Direction Artifact (DWI ONLY)'  the status field is 'Color_Artifact' and can have values "Good", "Fair", "Poor" or "Unacceptable"
2. A free form text comment that can be entered for that comment type (these go into column `Comment` of table `feedback_mri_comments`). On the feedback panel, you enter these in the various multi-line text fields, one for each comment type.
3. Various predefined comments associated to that comment type (these are defined in`feedback_mri_predefined_comment_types`). These are the numerous checkboxes that you find on the feedback panel. For example, for comments of type 'Dominant Direction Artifact (DWI ONLY)', you have predefined comments 'red artifact', 'green artifact' and blue artifact'. These are either turned on of off via a checkbox to indicate whether that predefined comment applies to that comment type.

This PR will ensure that:

- anything in category 1 will be named `<statusFieldName>_<scan_type>`, where `<scan_type>` is the type of scan the data dictionary entry is associated with. For example the status field name for comment type `Dominant Direction Artifact (DWI ONLY)` is `Color_Artifact` so the data dictionary entry for this entity when it is associated to scans of type `t1w` will be named `Color_Artifact_t1w`.
- anything in category 2 will be named `Comment_<statusFieldName>_<scan_type>`, where `<scan_type>` is the type of scan the data dictionary entry is associated with. For example the status field name for comment type `Dominant Direction Artifact (DWI ONLY)` is `Color_Artifact` so the data dictionary entries for these types of comments when they relate to scans of type `t1w` will be `Comment_Color_Artifact_t1w`.
- anything in category 3 will be named `PredefinedComment_<CommentText>_<scan_type>`, where `<scan_type>` is the type of scan the data dictionary entry is associated with. For example there is currently a predefined comment on the raisin bread database with text `red artifact`. The data dictionary entries for these types of predefined comments when they relate to scans of type `t1w` will be `PredefinedComment_red artifact_t1w`.

Note: this PR also corrects the contents of table `feedback_mri_comments` on raison bread: all records in that table either have to have a non NULL `PredefinedCommentID` or a non NULL `Comment` (or both).

#### Testing instructions (if applicable)

Try to run `tools/CouchDB_MRI_Imported.php` on the raisin bread data setr (or your project's data set). Make sure there are no errors and that the data dictionary entries are named correctly.
